### DISCUSSION TITLE
Convert into 3D photo/video like on Facebook?

### DISCUSSION THREAD

**Discussion by Jonseed** [thygate/stable-diffusion-webui-depthmap-script#14] [Q&A]
Does anyone know of a way to locally convert an image and depth map into a 3D photo/video like those you see on Facebook? I've seen some GitHub repos that will do this (like [this one](https://github.com/vt-vl-lab/3d-photo-inpainting)), but they are quite complex to set up, and require quite a lot of dependencies. I haven't been able to get any of them working on Windows. It would be cool if there was a tool that easily took an image and depth map like those this script produces, and produced a 3D photo/videos.

**Comment by kalkal11** [comment]
if you want the easy mode of using this, have a look at a program called visions of chaos but there are some prerequisites to get it running.

Funny you ask because I also asked if thygate might consider making an extension for that too earlier :)

**Reply by thygate** [reply]
I will definitely look into this later, I'm eager to see the `Inpainted 3D mesh` output myself.

**Reply by Jonseed** [reply]
What method or mode will turn an image and depth map into a 3D photo/video in Visions of Chaos?

**Reply by Jonseed** [reply]
Is it 3D Photo Inpainting?

**Reply by kalkal11** [reply]
@jonseed yes but it does the whole process automatically. Not quite what you're asking for though as it produced a pre-baked video. You'll want to edit the settings buried in the voc user folder to change the resolution to something higher. The config specifies longest side resolution which you'll need to increase

**Reply by Jonseed** [reply]
Hmm, yeah, I'm looking for something simpler, like Facebook, where you just drop in an image and depth map, and it makes a 3D photo or video for you.

**Reply by kalkal11** [reply]
@Jonseed how would you present this to people if not with facebook?

The VOC tool does produce a video for you, but there would be little point of producing an accelerometer based 3D photo without a form of delivery/viewing

**Reply by Jonseed** [reply]
@caacoe maybe a viewer needs to come with it then, a utility that will move the image around with mouse movement. Maybe it could even be viewed in a browser with some JavaScript like this: 

https://github.com/arturbien/3D-image

**Reply by Jonseed** [reply]
Or you could use this:
https://github.com/panrafal/depthy

**Reply by Jonseed** [reply]
You can actually use Depthy here. Just upload an image and its depth map:

http://depthy.stamina.pl/

It actually works quite well with the images and depth maps produced by this script!

**Reply by thygate** [reply]
>Depthy / 3D-image

Those are definitely nice options for viewing !

I will be adding a section on Viewing to the readme later.

**Reply by Jonseed** [reply]
@thygate I bet you could integrate the depthy repo with your script, or expand it into an extension, turn it into a whole tab in the UI.
https://github.com/panrafal/depthy

**Reply by thygate** [reply]
> @thygate I bet you could integrate the depthy repo with your script, or expand it into an extension, turn it into a whole tab in the UI. https://github.com/panrafal/depthy

I was thinking in the same direction..

**Reply by Jonseed** [reply]
@thygate also, you might include an option in the script to invert the depth map, as Depthy expects the reverse as you have it, black is near, white is far away.

**Reply by thygate** [reply]
> @thygate also, you might include an option in the script to invert the depth map, as Depthy expects the reverse as you have it, black is near, white is far away.

now included in v0.1.6

**Comment by thygate** [comment]
>Layered Depth Inpainting
>We propose a method for converting a single RGB-D input image into a 3D photo, i.e., a multi-layer representation for novel view synthesis that **_contains hallucinated color and depth structures in regions occluded in the original view_**. 

That's very interesting, I'll have to look into that.

I have threejs and Unity3D projects that display on lookingglass and SteamVR devices. (wip)
It's always the same technique of rendering a texture on a (subdivided) plane with a displacement modfier. You could do the same in blender for instance. This is already good when viewed straight-on on a 3D display, but as soon as you start moving the camera things fall apart quickly.

Layered Depth Inpainting looks like a good upgrade .. I will definitely look into it when i find some time.
Thanks for sharing and good tip !

**Reply by dangerweenie** [reply]
this is a popular technique in Nuke and After Effects as well for doing background with paralax (i.e. storm clouds moving in FG and BG, etc)
